Makefile

```diff
@@ -1,30 +1,73 @@
+MAKEFLAGS += --silent
 GOLANGCI_LINT_VERSION = $(shell head -n 1 .golangci.yml | tr -d '\# ')
 TMPDIR ?= /tmp
+K6_DEV_TOOLS_IMAGE = k6-dev-tools
 
-all: build
+# TODO: check if the image exists and if not suggest to run build-k6-dev-tools
+# TODO: implement validation if tool inside isn't outdated
+# TODO: a better cache key (maybe a image id)
+define run_k6_tools
+	@mkdir -p $(TMPDIR)/k6-dev-cache-$(GOLANGCI_LINT_VERSION)
+	@docker run --rm -t \
+		--user "$(shell id -u $(USER))" \
+		-v $(TMPDIR)/k6-dev-cache-$(GOLANGCI_LINT_VERSION):/golangci-cache \
+		--env "GOLANGCI_LINT_CACHE=/golangci-cache" \
+		-v $(shell pwd):/app \
+		-w /app $(K6_DEV_TOOLS_IMAGE) \
+		$(1)
+endef
+
+all: build check
+
+## help: Prints a list of available build targets.
+help:
+	echo "Usage: make <OPTIONS> ... <TARGETS>"
+	echo ""
+	echo "Available targets are:"
+	echo ''
+	sed -n 's/^##//p' ${PWD}/Makefile | column -t -s ':' | sed -e 's/^/ /'
+	echo
+	echo "Targets run by default are: `sed -n 's/^all: //p' ./Makefile | sed -e 's/ /, /g' | sed -e 's/\(.*\), /\1, and /'`"
 
-build :
+## clean: Removes any previously created build artifacts.
+clean:
+	rm -f ./k6
+
+## build: Builds the 'k6' binary.
+build:
 	go build
 
-format :
-	find . -name '*.go' -exec gofmt -s -w {} +
+## format: Applies Go formatting to code.
+format:
+	$(call run_k6_tools,gofumpt -w .)
 
-ci-like-lint :
-	@docker run --rm -t -v $(shell pwd):/app \
-		-v $(TMPDIR)/golangci-cache-$(GOLANGCI_LINT_VERSION):/golangci-cache \
-		--env "GOLANGCI_LINT_CACHE=/golangci-cache" \
-		-w /app golangci/golangci-lint:$(GOLANGCI_LINT_VERSION) \
-		make lint
+## fix: Runs golangci-lint with the verson that used inside the CI.
```

Comment:
Copy/paste inheritance 😉.  Should substitute "fix" with "lint" to match the target name.

Reply:
:facepalm: